docs/rules/no-disallowed-lwc-imports.md

```diff
@@ -0,0 +1,41 @@
+## Prevent importing restricted APIs from the "lwc" package (no-disallowed-lwc-imports)
+
+Restricts importing disallowed modules from the `lwc` package. It's recommended to only use the "safe" APIs allowed by this rule.
+
+This rule also disallows default imports as well as bare imports and exports – only importing explicitly named modules is allowed.
+
+### Rule details
+
+Examples of **incorrect** code:
+
+```js
+import { SecretApiYouShouldNotUse } from 'lwc';
+```
+
+```js
+import * as lwc from 'lwc';
+```
+
+```js
+import lwc from 'lwc';
+```
+
+```js
+import 'lwc';
+```
+
+```js
+export * from 'lwc';
+```
```

Comment:
Incidentally, our current Babel validator allows `export * from 'lwc'` and `export * as foo from 'lwc'`. This rule fixes that.